lua/lspconfig/server_configurations/terraformls.lua

```diff
@@ -1,10 +1,19 @@
 local util = require 'lspconfig.util'
 
+local language_id_mapping = {
+  tf = 'terraform',
+  tfvars = 'terraform-vars',
```

Comment:
What is not clear to me is how/where does `*.tf` get mapped to `tf` and `*.tfvars` to `tfvars`, i.e. pattern to filetype. Is this logic built into NeoVim somewhere? If not, then we'd need to do mapping based on patterns I think?

Reply:
filetype is detected by neovim internals
![image](https://user-images.githubusercontent.com/17252601/210773604-06a514f0-2906-456a-a33a-07c5d8a54e59.png)

![image](https://user-images.githubusercontent.com/17252601/210773741-fc137074-236f-4afc-b27f-3a72057ca89b.png)

Reply:
ok i realized we don't need it as it's the default behaviour:

> {get_language_id} function(bufnr, filetype) -> language ID as string. Defaults to the filetype.
https://neovim.io/doc/user/lsp.html

Reply:
Thank you for the context!

The only potential concern then is that "TF Mud files" will be sent to the Terraform LS too, just because these use the same `tf` filetype AFAICT: https://github.com/neovim/neovim/blob/5b9980f01eb021b0d84f540a6618f94ad2727153/runtime/filetype.vim#L1944-L1945
but I guess that's not a _new_ problem, esp. if the default mapping was in place before? 🤷🏻